src/Utilities/GenerateBatchVmSkus.cs

```diff
@@ -0,0 +1,163 @@
+                    List<ComputeResourceSku>? skus = null;
+                    foreach (var vm in vms)
+                    {
+                        if (regionsForVm.ContainsKey(vm))
+                        {
+                            regionsForVm[vm].Add(region);
+                        }
+                        else
+                        {
+                            regionsForVm[vm] = new HashSet<string>() { region };
+                        }
+
+                        if (!sizeForVm.ContainsKey(vm))
+                        {
+                            if (sizes is null)
+                            {
+                                sizes = await subscription.GetVirtualMachineSizesAsync(region).ToListAsync();
+                            }
+                            sizeForVm[vm] = sizes.FirstOrDefault(vmsize => vmsize.Name.Equals(vm));
+                        }
+
+                        if (!skuForVm.ContainsKey(vm))
+                        {
+                            if (skus is null)
+                            {
+                                skus = await subscription.GetComputeResourceSkusAsync(region).ToListAsync();
+                            }
+                            skuForVm[vm] = skus.FirstOrDefault(sku => sku.Name == vm);
+                        }
+                    }
+                    Console.WriteLine($"{region} supportedSkuCount:{vms.Count}");
+                }
+                catch (RequestFailedException e)
+                {
+                    Console.WriteLine($"No skus supported in {region}. {e.ErrorCode}");
+                }
+            }
+            
+            var batchSupportedVmSet = regionsForVm.Keys.ToList();
+            Console.WriteLine($"Superset supportedSkuCount:{batchSupportedVmSet.Count}");
+
+            var batchVmInfo = batchSupportedVmSet.Select((s) =>
+            {
+                var sizeInfo = sizeForVm[s];
+                var sku = skuForVm[s];
+                if (sizeInfo is not null)
+                {
+                    var generationList = new List<string>();
+                    var generation = sku?.Capabilities.Where(x => x.Name.Equals("HyperVGenerations")).FirstOrDefault()?.Value;
+
+                    if (generation is not null)
+                    {
+                        generationList = generation.Split(",").ToList();
+                    }
+
+                    int.TryParse(sku?.Capabilities.Where(x => x.Name.Equals("vCPUsAvailable")).FirstOrDefault()?.Value, out var vCpusAvailable);
+
+                    return new VirtualMachineInformation()
+                    {
+                        MaxDataDiskCount = sizeInfo.MaxDataDiskCount,
+                        MemoryInGB = ConvertMiBToGiB(sizeInfo.MemoryInMB ?? 0),
+                        VCpusAvailable = vCpusAvailable,
+                        ResourceDiskSizeInGB = ConvertMiBToGiB(sizeInfo.ResourceDiskSizeInMB ?? 0),
```

Comment:
ResourceDiskSizeInGiB ?  Also how could this ever be 0?

Reply:
It won't ever be 0, the null-coalesce is just there to cast int? to int before doing math.